src/UhdmAst.cpp

```diff
@@ -1264,30 +1259,238 @@ AstVar* process_param_assign(vpiHandle obj_h, UhdmShared& shared) {
+        if (shared.visited_types.find(object) != shared.visited_types.end()) {
+            UINFO(6, "Object " << objectName << " was already visited" << std::endl);
+            return nullptr;
+        }
+
+        shared.visited_types[object] = objectName;
+
+        // Use bare name for typespec itself, hierarchy was stored above
+        auto pos = objectName.rfind("::");
+        if (pos != std::string::npos) objectName = objectName.substr(pos + 2);
+
+        // VSigning below is used in AstStructDtype to indicate
+        // if packed or not
+        VSigning packed;
+        if (vpi_get(vpiPacked, obj_h)) {
+            packed = VSigning::SIGNED;
+        } else {
+            packed = VSigning::UNSIGNED;
+        }
+        auto* struct_dtype = new AstStructDType(new FileLine("uhdm"), packed);
+
+        vpiHandle member_itr = vpi_iterate(vpiTypespecMember, obj_h);
+        while(vpiHandle member_obj = vpi_scan(member_itr)) {
+            AstMemberDType* memberp = process_typespec_member(member_obj, shared);
+            if (memberp != nullptr)
+                struct_dtype->addMembersp(memberp);
+        }
+
+        auto* dtype = new AstDefImplicitDType(new FileLine("uhdm"), objectName, nullptr,
+                                              VFlagChildDType(), struct_dtype);
+        return dtype;
+    }
+    case vpiPackedArrayTypespec: {
+        vpiHandle index_typespec_h = vpi_handle(vpiIndexTypespec, obj_h);
+        const unsigned int index_type = vpi_get(vpiType, index_typespec_h);
+        AstBasicDTypeKwd typeKwd = get_kwd_for_type(index_type);
+        AstRange* rangeNodep = nullptr;
+        visit_one_to_many({vpiRange}, obj_h, shared,
+                          [&](AstNode* node) { rangeNodep = reinterpret_cast<AstRange*>(node); });
+        auto* dtypep = new AstBasicDType(new FileLine("uhdm"), typeKwd);
+        dtypep->rangep(rangeNodep);
+        return dtypep;
+    }
+    case vpiUnsupportedTypespec: {
+        v3info("\t! This typespec is unsupported in UHDM: " << file_name << ":" << currentLine);
+        // Create a reference and try to resolve later
+        return get_referenceNode(objectName);
+    }
+    }
+    return nullptr;
+}
+
+AstNode* process_typedef(vpiHandle obj_h, UhdmShared& shared) {
+    const unsigned int type = vpi_get(vpiType, obj_h);
+    if (type == UHDM::uhdmimport) {
+        // imports are under vpiTypedef nodes, but they are not defined types 
+        return process_uhdm_import(obj_h, shared);
+    }
+    
+    std::string objectName = get_object_name(obj_h);
+
+    auto pos = objectName.rfind("::");
+    if(pos != std::string::npos)
+        objectName = objectName.substr(pos + 2);
+
+    AstNodeDType* refp = nullptr;
+    if (vpiHandle alias_h = vpi_handle(vpiTypedefAlias, obj_h)) {
+        refp = getDType(alias_h, shared);
+    }
+    else {
+        refp = reinterpret_cast<AstNodeDType*>(process_typespec(obj_h, shared));
+        if (refp == nullptr)
```

Comment:
Yes. When I removed that `if`, the error during opentitan verilation was thrown: `%Error: Internal Error: ../V3Ast.cpp:344: Null item passed to setOp1p`. 
I will check if it is possible to handle that case instead of returning `nullptr`